### DISCUSSION TITLE
Would love to have a `--clean` option when running a program directly

### DISCUSSION THREAD

**Discussion by mortax** [grain-lang/grain#1601] [Ideas]
Currently, if i run a program directly, i.e.:

```
$ grain thing.gr
```

I am left with `thing.gr.wasm` and a `target` in the source directory:

```
$ ls
thing.gr
$ grain thing.gr 
$ ls
target/        thing.gr       thing.gr.wasm
```

I would love to be able to run a program without retained artifacts, e.g.:

```
$ ls
thing.gr
$ grain --clean thing.gr 
$ ls
thing.gr
```

**Comment by spotandjake** [comment]
Building on this it would be cool if it could write all the files it needs to a virtaul fs so you can run things in a read only environment. Disadvantage to this is I think you lose the build cache's right so builds rebuilds are slower but maybe grain could boodle to a temporary dir in this case.

It doesn't seem like a priority though what I do is I make a bash script at the root of my project that runs grain no the entry file and then cleans the artifacts after. Alternatively you might consider hiding the files from your ide.

**Reply by phated** [reply]
This is off-topic and covered in https://github.com/grain-lang/grain/issues/242

**Comment by phated** [comment]
@mortax This is unfortunately extremely difficult with the current way the compiler is built and how caching works. We plan to support this once we have a proper build tool (codenamed Silo) where we'll be tracking each file compiled.
